strawberryfields/program.py

```diff
@@ -526,13 +526,13 @@ def assert_max_number_of_measurements(self, device):
         num_pnr, num_homodyne, num_heterodyne = 0, 0, 0
 
         try:
-            max_pnr = device.modes["max"]["pnr"]
-            max_homodyne = device.modes["max"]["homodyne"]
-            max_heterodyne = device.modes["max"]["heterodyne"]
+            max_pnr = device.modes["pnr_max"]
+            max_homodyne = device.modes["homodyne_max"]
+            max_heterodyne = device.modes["heterodyne_max"]
         except (KeyError, TypeError) as e:
             raise KeyError(
                 "Device specification must contain an entry for the maximum allowed number "
-                "of measurments. Have you specified the correct target?"
+                "of measurements. Have you specified the correct target?"
```

Comment:
Are we sure that, "Have you specified the correct target?" is an accurate suggestion? It seems to me that users would only see this message when there is a bug in the SF-XC interface!

Reply:
Hmm... you're probably right. This message can be improved.